service/worker/scanner/README.md

```diff
@@ -0,0 +1,247 @@
+- Collect aggregated metrics
+- Periodically flush details to a blob-store
+
+# Some other files involved that seemed interesting to link
+
+- [/common/reconciliation/invariant/types.go](../../../common/reconciliation/invariant/types.go)
+  - `Name` entries name and refer to individual `Invariant` implementations.
+  - `Collection` entries define... categories of sorts, which are used elsewhere to configure which `Invariant`s run.
+- [/service/worker/scanner/executions/types.go](executions/types.go)
+  - `Collection` selections elsewhere are converted to `Invariant`s, based on the scanner type
+    - Timer scanners are handled in [scanner/timers/timers.go](timers/timers.go) rather than in here
+- [/service/worker/scanner/executions/concrete_execution.go](executions/concrete_execution.go)
+  - this and [current_execution.go](executions/current_execution.go) define a `[ExecutionType]ExecutionConfig` func that can override which `Collection`s are run in a scanner.
+
+# Wishlist
+
+- Collections and how config is serialized -> checked against hardcoded collections -> re-serialized repeatedly
+  make it impossible to add invariants in external code.
+  - There are _many_ ways this could be made both simpler and more flexible.
+  - This is also a major reason it's difficult to e.g. add loggers or change APIs.
+    They're repeatedly treated identically, then hardcoded, then identical, etc and customizations need to be threaded through all stages.
+  - The config-reading activity at the start of the workflows is a prime example of this.
+    - It could have just been a list of invariant names, and all this logic could have been handled by `encoding/json` plus a duplicate check in the invariant manager or something.
+    - Instead it introduces custom serialization far away from the type, uses iotas and enumer codegen and switch statements which share many logically-incompatible types,
+      has a `map[string]string` pulling double duty as a set (it is at least uniquely typed!), doesn't complain about passing incorrect keys (a wrong-type invariant is ignored),
+      uses multiple different and disconnected kinds of strings and keys and values to control if an invariant is included or not, and is spread across several files.
+- "Don't run the workers if they are not configured" seems... maybe not ideal?
+  - Some benefits, some downsides, but definitely more complicated.
+  - Maybe just run these workers all the time?  Workflows/etc can be canceled if you want to stop them, the current setup only _pauses_ them (with essentially no timeout).
+- "Scan everything then fix everything" has pretty clear downsides, not least of which is potentially-massive latency.
+  - Being _able_ to separate them is useful for verification, that should be kept.
+- Invariants (some anyway) are used elsewhere, which is part of why they're highly generic.
+  - I have not investigated this use of invariants.
+  - Might only be related to NDC replication?
```

Comment:
the idea of invariants more generally yeah, I think could be super useful in all sorts of areas, including history building too. But the idea of an invariant as it *might* Be expressed (a pure fucntion which takes a series of states and returns true or false) and its implementation here in the scanner where it's coupled with DB iterators is probably a detail for an entire deep-dive.